sdk/keyvault/azure-security-keyvault-keys/CMakeLists.txt

```diff
@@ -2,19 +2,41 @@
 # SPDX-License-Identifier: MIT
 
 cmake_minimum_required (VERSION 3.13)
+project(azure-security-keyvault-keys LANGUAGES CXX)
 
-# CodeCoverage will validate requirements
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../../cmake-modules")
+
+include(az_version)
 include(CodeCoverage)
+include(DefineTransportAdapter)
+include(doxygen_common)
+include(global_compile_options)
 
 add_library(azure-security-keyvault-keys
-    src/)
+    src/key_client.cpp
+    src/key_type.cpp
+    src/key_vault_key.cpp
+)
+
 target_include_directories (azure-security-keyvault-keys PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/inc> $<INSTALL_INTERFACE:include/azure-security-keyvault-keys>)
-target_link_libraries(azure-security-keyvault-keys azure-core)
-add_library(azure::keyvault::keys ALIAS azure-security-keyvault-keys)
+target_link_libraries(azure-security-keyvault-keys PUBLIC azure-security-keyvault-common)
 
-get_az_version("${CMAKE_CURRENT_SOURCE_DIR}/inc/azure/keyvault/keys/version.hpp")
+add_library(azure::keyvault::keys ALIAS azure-security-keyvault-keys)
```

Comment:
Just double checking, though I think the answer is "no", but this shouldn't be `SHARED`? I.e. all sources are compiled and statically linked?

Reply:
Default will do STATIC linking.  There is an env variable for CMake to link dynamically.
I think we currently have a bug for dynamic linking #1234